### PR TITLE
feat(extension): conventional commit w/ gitmoji

### PR DESCRIPTION
**Pull request by JeromeFitz**
## Description

Extension: **Conventional Comments**

Addition to prepare a [`conventional commit`](https://www.conventionalcommits.org) with the option of using [Gitmoji](https://gitmoji.dev) (emoji preference).

## Type of change

- Bug fix / improvement for somebody else's extension. Extension author: @krzysztofzuraw 

## Screencast

https://user-images.githubusercontent.com/3099369/147155782-fc63c992-531c-41dd-910f-cf7c0bde64cd.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

**Comment by JeromeFitz** [comment]
Before taking out of draft state for potential review, wanted to see if we could / should instead combine this with:

- `conventional-comments`
  - https://github.com/raycast/extensions/pull/94

We could add an option if the person is writing a `comment` or a `commit` and keep this to one extension.

Thoughts @krzysztofzuraw ? If this is going down an acceptable path the PR can be reworked to include both.

Unsure if adding `gitmoji` makes this too opinionated for your initial extension addition. 🙇🏻

**Comment by krzysztofzuraw** [comment]
Awesome work 👍🏻 . Maybe we can combine those two extensions into one with option to use emoji or not in preferences? @JeromeFitz

**Comment by JeromeFitz** [comment]
Sweet, sounds great @krzysztofzuraw!

@pernielsentikaer I should have time to rework this PR over the weekend to keep as one extension within Raycast with two commands.

**Conventional**
- `Insert Conventional Comment`
- `Insert Conventional Commit`

**Comment by pernielsentikaer** [comment]
Sound like a plan to merge them into one instead of having two similar extensions 😄

**Comment by Four-Velocity** [comment]
Hey there :wave:
Correct me if I'm wrong, but sometimes you have to combine several emojis in one commit. The example that first came to my mind is `:beers:` and any other emoji, but there are definitely more of them :thinking:.
So perhaps it makes sense to add smth like a plus button to add one more commit type? :smiley:

**Comment by fe9lix** [comment]
Hey guys, what's the status with this PR? (draft mode and last activity 28 days ago...)

**Comment by JeromeFitz** [comment]
👋🏻 @fe9lix , thanks for the ping.

- ♻️ refactored the `commit` part to be more streamlined
- 🚚️ moved it into the existing extension: `conventional-comments`

There are now two commands:

1. Insert Conventional Comment
2. Insert Conventional Commit 

image

I've left the initial `comment` as-is, if we want to add in gitmoji / emojis to that later we can. CC: @krzysztofzuraw 

Unsure if this extension should be renamed to `conventional`, however, was not sure how that may work with existing installs so left as-is.

@Four-Velocity I think that could be a cool idea for sure. Perhaps something that could be added in the future. 🆒

File: extensions/conventional-comments/README.md [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+#### Breaking Change
+
+If checkbox is selected will require additional notation.
+
+## Roadmap
+
+- [ ] Preview commit based on format selected
+- [ ] Custom formatting options (subject, breaking change, etc.)
+- [ ] Automatic updates from Gitmoji (?)
+- [ ] Issue Tracker integration (?)
```
Comment (JeromeFitz) [inline extensions/conventional-comments/README.md:46]: **thought**: 
May not be the right place for this, and can be removed. 🔥️

**Review by fe9lix — CHANGES_REQUESTED** [review]
Hey @JeromeFitz – I've noticed two things that should be addressed before we can merge:

- There's some slight flickering when initially rendering the commit form and as far as I can tell, this comes from the way the checkbox state is updated, leading to initial rendering with different states (additional form element shown, then form element hidden). Here's a way to fix this: Set the intitial state value for `isBreakingChange` to `false`. Add a controlled property value to the checkbox (`value={isBreakingChange}`), and in the change handler, do not flip the boolean but rather set the state to the current `value` parameter.

- There's no icon in the last dropdown item for "refactor(scope)" description – either all items should have an icon or none of them

**Comment by JeromeFitz** [comment]
Thanks for the review @fe9lix and detailed breakdown. 💯 

 ✅ Updated the checkbox to the proper standards. Removed the handler in preference to the [Controlled Checkbox behavior](https://developers.raycast.com/api-reference/user-interface/form#form.checkbox). (+ removed `ROADMAP` from README).

For the last dropdown `Format`, the (current) intent is to show the 3 formats available to compile the information being set in the fields above. The last one does not have an emoji, as the user is not selecting a format that displays one.

https://user-images.githubusercontent.com/3099369/147262074-5169f61d-5a0c-46ab-8f8a-92cbd1abc901.mov




In general `format` could probably be more intuitive to the end user. I will give it more thought, however, maybe removing the `format` field in preference to it being a configuration setting could reduce clutter and enable a better user experience when the `Breaking Change Info` appears as well.

🤔  Maybe a more stepped approach where the person can focus (quickly) on one thing at a time and rapidly move through.

**Review by fe9lix — APPROVED** [review]

**Pull request merged**